src/annotation/askui-annotation-step-reporter.ts

```diff
@@ -0,0 +1,80 @@
+import fs from "fs";
+import { Step, Reporter, Annotation, ReporterConfig } from "askui";
+import path from "path";
+import { JSDOM } from 'jsdom';
+
+export enum AnnotationLevel {
+  DISABLED = 'disabled',
+  ON_FAILURE = 'onFailure',
+  ALL = 'all',
+}
+
+export interface AnnotationReporterConfig extends ReporterConfig {
```

Comment:
@adi-wan-askui  Can this be solved with TypeScript Magic as the constructor looks really ugly.

Reply:
To be honest, I would simply not extend `ReporterConfig` but keep that internal to `AskUIAnnotationStepReporter` and set it based on how `annotationLevel` is set.